### PR TITLE
ens support

### PR DESCRIPTION
**Pull request by aquaticone**
Closes https://github.com/FortressFinance/issues/issues/79
Depends on #17 
Diff: https://github.com/FortressFinance/ui/compare/aquatic/providers...aquatic/ens

Changes:
- custom ens hook with mainnet provider
- fixed minor styling issue noticed while working on this
- added multicall3 contract addresses for fork networks

### CODE REVIEW

**Comment by vanillaHill** [comment]
Need to update with the previous PR.

File: src/components/Layout.tsx [hunk 1]
```diff
@@ -1,26 +1,19 @@
...
 import { useConnectWallet } from "@/store/connectWallet"
 
 import FortressBackground from "~/images/fortress-background.gif"
 import FortressLogo from "~/svg/fortress-logo.svg"
 
-const ConnectWalletButton = dynamic(
-  () => import("@/components/ConnectWallet/ConnectWalletButton"),
-  { ssr: false }
-)
-const NetworkSelector = dynamic(() => import("@/components/NetworkSelector"), {
```
Comment (aquaticone) [inline src/components/Layout.tsx:20]: Hydration errors can occur with components that rely on client state (i.e. wagmi), but these don't need it anymore. Removing dynamic imports allows the server to render the elements on the page, thus reducing layout shift when the client hydrates.

File: src/components/Vault/VaultTable.tsx [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 import { VaultProps } from "@/hooks/types"
 
-import Spinner from "@/components/Spinner"
-import VaultRow from "@/components/Vault/VaultRow"
 import {
   VaultTableHeader,
   VaultTableRow,
 } from "@/components/Vault/VaultTableNode"
 
+const VaultTableBody = dynamic(
```
Comment (aquaticone) [inline src/components/Vault/VaultTable.tsx:11]: Moving the dynamic import down to the VaultBody component to reduce layout shift a bit more. Headers will be on the page when it is loaded by the client.

File: src/components/WagmiProvider.tsx [hunk 3]
```diff
@@ -38,6 +38,12 @@ export const mainnetFork: Chain = {
       http: ["/api/anvil-mainnet"],
     },
   },
+  contracts: {
+    multicall3: {
```
Comment (aquaticone) [inline src/components/WagmiProvider.tsx:42]: More realistic for testing.

File: src/hooks/useMainnetEnsName.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+type FetchMainnetEnsNameResult = string | null
+
+async function fetchMainnetEnsName({
+  address,
+}: {
+  address: Address | undefined
+}): Promise<FetchMainnetEnsNameResult> {
+  if (!address) throw new Error("address is required")
+  const provider = new ethers.providers.AlchemyProvider(
```
Comment (aquaticone) [inline src/hooks/useMainnetEnsName.ts:63]: Bypassing wagmi providers here because this is the only request we need to make against mainnet.

**Review by vanillaHill — APPROVED** [review]

**Pull request merged**